include/rocksdb/table.h

```diff
@@ -86,6 +86,9 @@ struct BlockBasedTableOptions {
     // The hash index, if enabled, will do the hash lookup when
     // `Options.prefix_extractor` is provided.
     kHashSearch,
+
+    // A two-level index implementation. Both levels are binary search indexes.
+    kTwoLevelIndexSearch,
```

Comment:
Sure will add that.

My thinking is to allow enable/disable each feature separately to measure their impact in the benchmarks. That extends to the interleaving phase. I have not thought it through about the switches of how to enable/disable them though and am open to suggestions.